content/en/agent/autodiscovery/integrations.md

```diff
@@ -0,0 +1,541 @@
+
+Each tab in sections below shows a different way to apply Integration templates to a given container, available documented method are:
+
+* [Using a configuration file mounted within the Agent](?tab=file)
+* [Using Key-Value Store](?tab=keyvaluestore)
+* [Using Kubernetes Annotations](?tab=kubernetespodannotations)
+* [Using Docker Labels](?tab=dockerlabel)
+
+If you provide a template for the same Integration via multiple template sources, the Agent looks for templates in the following order (using the first one it finds):
+* Kubernetes annotations
+* Files
+
+**Note**: Some supported integrations requires additional steps for Autodiscovery to work: [Ceph][3], [Varnish][4], [Postfix][5], [Cassandra Nodetools][6], and [Gunicorn][7]. Contact [Datadog support][8] for assistance.
+
+## Configuration
+
+{{< tabs >}}
+{{% tab "File" %}}
+
+Storing templates as local files and mounting them inside the containerized Agent doesn't require an external service or a specific orchestration platform. The downside is that you have to restart your Agent containers each time you change, add, or remove templates. The Agent looks for Autodiscovery templates in the mounted `/conf.d` directory, which contains default templates for the following checks: [Apache][1], [Consul][2], [CouchDB][3], [Couchbase][4], [Elasticsearch][5], [Etcd][6], [Kubernetes_state][7], [Kube_dns][8], [Kube_proxy][9], [Kyototycoon][10], [Memcached][11], [Redis][12], [Riak][13].
+
+Since v6.2.0 (and v5.24.0), the default templates use the default port for the monitored software, instead of auto-detecting it. If you need to use a different port, provide a custom Autodiscovery template either in [Docker container labels](?tab=docker-labels) or [Kubernetes pod annotations](?tab=kubernetes-annotations).
+
+These integration templates may suit you in basic cases, but if you need to customize your Datadog Integration configurations: such as enabling extra options, using different container identifiers, or using Template Variables indexing, you have to write your own auto-configuration files and use them with the Datadog Containerized Agent. To do this:
+
+1. Create a `autodiscovery.d/` folder on your host.
+2. Add your custom auto-configuration named `<INTEGRATION_NAME>.yaml`file to this folder.
+3. Mount this directory into the containerized Agent `conf.d/` folder. Note that On Kubernetes your can add them [using ConfigMaps][14].
+
+An alternative approach would be to:
+
+1. Create a `autodiscovery.d/<CUSTOM_FILE_NAME>.yaml` file on your host.
+2. Mount this file inside the `conf.d/<INTEGRATION_NAME>.d/` folder of the containerized Agent.
+
+Your auto-configuration file should have the following format:
+
+```
+ad_identifier:
+  <INTEGRATION_AUTODISCOVEY_IDENTIFIER>
+
+init_config:
+  <INIT_CONFIG>
+
+instances:
+  <INSTANCES_CONFIG>
+```
+
+**Note**: You don't need to set up the `<INTEGRATIONS_NAME>` there, since the Agent inferred it from the file name directly.
```

Comment:
Not sure to understand the idea there. The folder name should be set to the integration name by hand when doing the mount.

Reply:
that's true, but it turns out there is a script when the Agent starts that does this automatically :)

It looks at all file, and depending on their name put them it the corresponding integrations configuration folder :)

Reply:
Good to know! so custom_file_name should be the name of the integration then?

Reply:
nop, `<INTEGRATIONS_NAME>` should be the name of the integration.

But after reading it i understand the confusion, i'm going to update the wording.

Reply:
🤔 can you point to the script that does this?

I'm alway using `<INTEGRATION_NAME.d/whatever.yaml`